### ISSUE TITLE
Require email for attendee

### ISSUE THREAD

**Issue by midzdotdev** [sebbo2002/ical-generator#344]
Just tried to submit a PR and failed.

I've tried to create events with attendees who have no email and it this library throws. I feel that the types for `ICalInternalAttendeeData` and other relevant types should indeed require `email`.

**Comment by james2mid** [comment]
It's when the Attendee is stringified and throws for a falsy email [on this line](https://github.com/sebbo2002/ical-generator/blob/c6e79f796c02a24430d2248f2fcf4be6d5d3c43b/src/attendee.ts#L531)

I think allowing constructing an attendee with a falsy email is confusing. I suppose this would also apply to the `email` setter 
accepting falsy values too.

**Comment by james2mid** [comment]
Awesome library btw! ❤️

**Comment by sebbo2002** [comment]
In fact this is true for all mandatory fields in ical-generator. The reason for this is that ical-generator has always allowed you to use the setters, in this case `email()` for example, and you are not required to include all the data with the constructor. This is currently absolutely valid code, for example:

```js
const attendee = event.createAttendee();
attendee.email('foo@bar.com');
```

I understand that this is a bit weird, but I hope that the exceptions is meaningful enough. For the next breaking version it is already planned that at least the mandatory fields have to be passed to achieve some security here.

**Comment by james2mid** [comment]
Yes thank makes perfect sense. I'm personally a big fan of immutable data structures and so recreating when a change is needed. In this case, there can never be an event or an attendee with a state which cannot be stringified to iCalendar.

It sounds like your moving towards that a little bit in the next breaking version you mentioned.

I wish you the best with this :)

**Comment by sebbo2002** [comment]
I hope so, thank you very much. I would close the ticket for now, okay?

**Comment by sebbo2002** [comment]
I just pushed a preview of `ical-generator` v6 to develop, which should fix this issue. I would be happy if you could test the new version and give me some feedback. You can install the version with `npm i ical-generator@next`. All changes in this release can be found [here](https://github.com/sebbo2002/ical-generator/wiki/Migration-Guide:-v5-%E2%86%92-v6).
